### PR TITLE
Enable checkout with regex

### PR DESCRIPTION
**Pull request by limor-gs**

### CODE REVIEW

File: clue/blueprint/cloudify-dev/operations/git.py [hunk 1]
```diff
@@ -103,6 +104,15 @@ def checkout(self, branch):
         versions_prefix = '::'
         default_branch = self.branch
         active_feature = self.active_feature
+
+        if '*' in branch:
+            branch = branch.replace('.', '\.')
+            branch_name = self._get_branch_name_from_regex(branch)
+            if branch_name:
```
Comment (mcouthon) [inline clue/blueprint/cloudify-dev/operations/git.py]: It's nicer to do:
` branch = str(branch_name) if branch_name else default_branch`

File: clue/blueprint/cloudify-dev/operations/git.py [hunk 2]
```diff
@@ -395,6 +405,29 @@ def _read_versions_file(self, versions_branch):
                     versions_branch)).stdout.strip()
         versions = yaml.safe_load(raw_versions)
         return versions.get('components', {})
+
+    def _get_branch_name_from_regex(self, branch):
+        branches = self.git_output('branch', '-a').stdout.strip().split('\n')
```
Comment (mcouthon) [inline clue/blueprint/cloudify-dev/operations/git.py]: Better to call `escape_ansi` once on git_output, and then split on '\n'

File: clue/blueprint/cloudify-dev/operations/git.py [hunk 3]
```diff
@@ -395,6 +405,29 @@ def _read_versions_file(self, versions_branch):
...
+        branches = [str(self._escape_ansi(item)).strip() for item in branches]
+        branch_names = [item.split("remotes/")[-1].split("origin/")[-1]
+                        for item in branches if re.search(branch, item)]
+        # remove duplicate branches remote and local
+        branch_name = list(set(branch_names))
```
Comment (mcouthon) [inline clue/blueprint/cloudify-dev/operations/git.py]: `list` here is unnecessary. Just do `set`

File: clue/blueprint/cloudify-dev/operations/git.py [hunk 4]
```diff
@@ -395,6 +405,29 @@ def _read_versions_file(self, versions_branch):
...
+        if len(branch_name) > 1:
+            ctx.logger.error('More than one branch with regex {0} found: {1}'
+                             .format(branch, branch_name))
+            return None
+        elif len(branch_name) == 1:
+            branch_name = ''.join(branch_name[0])
```
Comment (mcouthon) [inline clue/blueprint/cloudify-dev/operations/git.py:418]: If it's a set you can do `str(next(branch_name))`

**Review by mcouthon — CHANGES_REQUESTED** [review]

File: clue/blueprint/cloudify-dev/operations/git.py (hunk 3 above)
Comment (limor-gs) [reply clue/blueprint/cloudify-dev/operations/git.py]: doesn't work

**Review by mcouthon — APPROVED** [review]

**Pull request merged**